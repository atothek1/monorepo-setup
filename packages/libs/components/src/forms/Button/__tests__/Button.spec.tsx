import React from "react";
import { render, userEvent } from "@mono/testing";
import { Button } from "../index";

const testId = "testid.button";

describe( "Button snapshots", () => {
    it( "should render default", () => {
        const { getByTestId } = render( <Button testId={ testId }>test</Button> );
        expect( getByTestId( testId ) ).toMatchSnapshot();
    } );

    it( "should render as type submit", () => {
        const { getByTestId } = render( <Button testId={ testId } type="submit">test</Button> );
        expect( getByTestId( testId ) ).toMatchSnapshot();
    } );

    it( "should render as type reset", () => {
        const { getByTestId } = render( <Button testId={ testId } type="reset">test</Button> );
        expect( getByTestId( testId ) ).toMatchSnapshot();
    } );

    it( "should render disabled", () => {
        const { getByTestId } = render( <Button testId={ testId } isDisabled>test</Button> );
        expect( getByTestId( testId ) ).toMatchSnapshot();
    } );
} );

describe( "Button behaviour", () => {

    it( "should call internal noop handler", () => {
        const handleClick = jest.fn();
        const { getByTestId } = render( <Button testId={ testId }>test</Button> );
        expect( handleClick ).not.toBeCalled();
        userEvent.click( getByTestId( testId ) );
    } );

    it( "should call passed in handler", () => {
        const handleClick = jest.fn();
        const { getByTestId } = render( <Button testId={ testId } onClick={ handleClick }>test</Button> );
        expect( handleClick ).not.toBeCalled();
        userEvent.click( getByTestId( testId ) );
        expect( handleClick ).toBeCalledTimes( 1 );
    } );

    it( "should not call passed in handler as button is disabled", () => {
        const handleClick = jest.fn();
        const { getByTestId } = render( <Button testId={ testId } isDisabled onClick={ handleClick }>test</Button> );
        userEvent.click( getByTestId( testId ) );
        expect( handleClick ).not.toBeCalled();
    } );
} );
