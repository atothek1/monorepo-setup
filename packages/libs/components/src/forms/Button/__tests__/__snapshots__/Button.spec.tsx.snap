// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots should render as type reset 1`] = `
.c0 {
  border: 0;
  outline: none;
  text-align: center;
  color: white;
  background-color: dodgerblue;
  padding: 0.8rem;
  border-radius: 4px;
}

<button
  class="Button___StyledStyledButton-w4d9ie-0 c0"
  data-testid="testid.button"
  type="reset"
>
  test
</button>
`;

exports[`Button snapshots should render as type submit 1`] = `
.c0 {
  border: 0;
  outline: none;
  text-align: center;
  color: white;
  background-color: dodgerblue;
  padding: 0.8rem;
  border-radius: 4px;
}

<button
  class="Button___StyledStyledButton-w4d9ie-0 c0"
  data-testid="testid.button"
  type="submit"
>
  test
</button>
`;

exports[`Button snapshots should render default 1`] = `
.c0 {
  border: 0;
  outline: none;
  text-align: center;
  color: white;
  background-color: dodgerblue;
  padding: 0.8rem;
  border-radius: 4px;
}

<button
  class="Button___StyledStyledButton-w4d9ie-0 c0"
  data-testid="testid.button"
  type="button"
>
  test
</button>
`;

exports[`Button snapshots should render disabled 1`] = `
.c0 {
  border: 0;
  outline: none;
  text-align: center;
  color: white;
  background-color: dodgerblue;
  padding: 0.8rem;
  border-radius: 4px;
}

<button
  class="Button___StyledStyledButton-w4d9ie-0 c0"
  data-testid="testid.button"
  disabled=""
  type="button"
>
  test
</button>
`;
